### PR TITLE
Remove the attributes class and use directly a map.

### PR DESCRIPTION
**Pull request by bogdandrutu**
The reason is that before we were forced to do an extra copy of the map when we implement AddAttributes (once in the Attributes class and once in the implementation).

### CODE REVIEW

**Comment by adriancole** [comment]
Sgtm

File: core/src/main/java/com/google/instrumentation/trace/AttributeValue.java [hunk 1]
```diff
@@ -13,30 +13,51 @@
...
+
 import com.google.common.base.MoreObjects;
 import com.google.common.base.Objects;
 import javax.annotation.Nullable;
 import javax.annotation.concurrent.Immutable;
 
 /**
- * A class that represents all the possible values for an attribute in {@link Attributes}. A
- * attribute can have 3 types of values: {@code String}, {@code Boolean} or {@code Long}.
+ * A class that represents all the possible values for an attribute . A attribute can have 3 types
```
Comment (a-veitch) [inline core/src/main/java/com/google/instrumentation/trace/AttributeValue.java]: grammar nit: s/A attribute/An attribute/

File: core/src/test/java/com/google/instrumentation/trace/BlankSpanTest.java [hunk 2]
```diff
@@ -30,13 +32,13 @@ public void hasInvalidContextAndDefaultSpanOptions() {
 
   @Test
   public void doNotCrash() {
+    Map<String, AttributeValue> attributes = new HashMap<String, AttributeValue>();
+    attributes.put(
```
Comment (a-veitch) [inline core/src/test/java/com/google/instrumentation/trace/BlankSpanTest.java]: Maybe add one test with multiple attributes, different types.

**Comment by bogdandrutu** [comment]
Updated. Waiting for tests to pass.

**Pull request merged**